### PR TITLE
Rearrange categries 4 col

### PR DESCRIPTION
**Pull request by invisiblemask**
# Rearrangement of Categories card from 3 columns to 4 

_This PR fixes the issue of replacing 3 columns with 4 for the categories section, this makes images that are pixelated clearer_

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/3065

### CODE REVIEW

File: src/components/Categories/CategoriesCard.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export default function CategoriesCard({ wiki }: CategoriesCardProps) {
           {shortenText(wiki.title, 30)}
         </h1>
         <p className="dark:text-alpha-800 text-gray-600 text-xs lg:text-sm">
-          {shortenText(wiki.summary, 80)}
+          {shortenText(wiki.summary, 50)}
```
Comment (kesar) [inline src/components/Categories/CategoriesCard.tsx:48]: not big deal but code would look much better if we extract these 50, 80, 30 magic numbers into constants on top of the component and use it there 👍🏻

File: src/components/Landing/CategoriesList.tsx [hunk 2]
```diff
@@ -78,14 +78,14 @@ const CategoriesList = ({ categories }: CategoriesListProps) => {
...
                     (category) =>
-                      category?.id === allCategory.id && (
+                      category.id === allCategory.id && (
                         <div
                           key={category.id}
-                          className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-6"
+                          className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-4 gap-6"
                         >
-                          {category.wikis.slice(0, 6).map((wiki) => (
+                          {category.wikis.slice(0, 8).map((wiki) => (
```
Comment (kesar) [inline src/components/Landing/CategoriesList.tsx:88]: same wth this number, probably extract and const in the beginning to easily spot and replace

**Comment by kesar** [comment]
great work, just some small suggestions

File: src/components/Categories/CategoriesCard.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ export default function CategoriesCard({ wiki }: CategoriesCardProps) {
       href={`/wiki/${wiki.id}`}
       className="flex flex-col gap-2 border dark:border-gray-700 border-gray-200 rounded-xl pb-2 hover:scale-105 transition-transform duration-300 ease-in-out"
     >
-      <div className="h-80">
+      <div className="aspect-auto">
```
Comment (kesar) [inline src/components/Categories/CategoriesCard.tsx:34]: not sure how this work, but lets ensure this doesnt make things weird in some of the different screen sizes

File: src/components/Categories/CategoriesCard.tsx (hunk 1 above)
Comment (invisiblemask) [reply src/components/Categories/CategoriesCard.tsx:48]: Okay sir

File: src/components/Categories/CategoriesCard.tsx (hunk 3 above)
Comment (invisiblemask) [reply src/components/Categories/CategoriesCard.tsx:34]: I am already testing that

**Review by Simplypheyie96 — CHANGES_REQUESTED** [review]
1. it's for home and categories page, so we can't close this out yet.
2. There is inconsistency with the placement of the article author and i am just noticing that the "date edited is missing".
image

3. just like the design, we would have to adjust the font size of the wiki title and subtitles on the card since the cards are smaller now.

**Comment by invisiblemask** [comment]
> 1. it's for home and categories page, so we can't close this out yet.
> 
> 2. There is inconsistency with the placement of the article author and i am just noticing that the "date edited is missing".
> 
> image
> 
> 
> 
> 3. just like the design, we would have to adjust the font size of the wiki title and subtitles on the card since the cards are smaller now.

Do you mean changing the wiki cards in categories page to 4 columns too?

The author name is not consistent because the wiki summary length is very small and that's because it's a test wiki, the summary length for the actual wikis are longer, but I will adjust that in case there's any scenario where we would have short wiki summary length and also add the date edited too

I will check the design and update the font sizes

**Comment by Simplypheyie96** [comment]
> Do you mean changing the wiki cards in categories page to 4 columns too?

It was added as part of the issue. @Softdev1 please confirm again if we are changing the categories page into 4 columns.
image

**Comment by Softdev1** [comment]
> > Do you mean changing the wiki cards in categories page to 4 columns too?
> 
> It was added as part of the issue. @Softdev1 please confirm again if we are changing the categories page into 4 columns. image

Yes we are doing it on both the homepage and the category page too

**Comment by Simplypheyie96** [comment]
> Yes we are doing it on both the homepage and the category page too

To avoid pr conflict, @Yomzy2223  can work on the the categories page since he is already working on the categories page.

**Review by Simplypheyie96 — APPROVED** [review]

**Comment by Tolu-Mals** [comment]
When the user doesn't have a username, it should show the address instead of being blank

Screenshot 2024-08-20 at 9 39 40 AM

Reference - Activity Card
Screenshot 2024-08-20 at 9 40 16 AM

**Review by Softdev1 — APPROVED** [review]

**Review by Tolu-Mals — APPROVED** [review]

**Pull request merged**